src/components/donate/Stripe.js

```diff
@@ -152,103 +155,103 @@ const PaymentForm = (props) => {
               frequency={frequency}
             />
           </Grid>
+        )}
 
-          <Grid item xs={12} sm={compact ? 12 : 6}>
-            <NameField
-              form={form}
-              classes={classes}
-              name="firstname"
-              label={t("First name")}
-              autoComplete="given-name"
-            />
-          </Grid>
-          <Grid item xs={12} sm={compact ? 12 : 6}>
-            <NameField
-              form={form}
-              classes={classes}
-              name="lastname"
-              label={t("Last name")}
-              autoComplete="family-name"
-            />
-          </Grid>
+        <Grid item xs={12} sm={compact ? 12 : 6}>
+          <NameField
+            form={form}
+            classes={classes}
+            name="firstname"
+            label={t("First name")}
+            autoComplete="given-name"
+          />
+        </Grid>
+        <Grid item xs={12} sm={compact ? 12 : 6}>
+          <NameField
+            form={form}
+            classes={classes}
+            name="lastname"
+            label={t("Last name")}
+            autoComplete="family-name"
+          />
+        </Grid>
+        <Grid item xs={12}>
+          <Controller
+            control={control}
+            name="email"
+            // KISS for email validation - something@something\.something
+            rules={{ required: true, pattern: /^(.+)@(.+)\.(.+)$/ }}
+            render={({ onChange, onBlur, value }) => (
+              <LayoutTextField
```

Comment:
I had to fallback to the core MUI component because the validation caused too many re-renders. My comment about it is probably in the wrong place:

```
//import TextField from "../TextField";
// We can't use the goodies of our material ui wrapper, because it triggers
// too many redraw and sometimes clear the stripe field (credit cards when it shouldn't)
```

Reply:
Haha - actually that's your comment!